core/frontend/src/views/ExtensionManagerView.vue

```diff
@@ -0,0 +1,212 @@
+import back_axios from '@/utils/api'
+
+import { ExtensionData } from '../types/kraken'
+
+const API_URL = '/kraken/v1.0'
+
+const notifier = new Notifier(kraken_service)
+
+export default Vue.extend({
+  name: 'ExtensionManagerView',
+  components: {
+    ExtensionCard,
+    ExtensionModal,
+  },
+  data() {
+    return {
+      tab: 0,
+      show_dialog: false,
+      installed_extensions: [] as ExtensionData[],
+      selected_extension: null as (null | ExtensionData),
+      manifest: [] as ExtensionData[],
+      dockers_fetch_done: false,
+    }
+  },
+  mounted() {
+    this.fetchManifest()
+    this.fetchInstalledExtensions()
+  },
+  methods: {
+    async fetchManifest(): Promise<void> {
+      await back_axios({
+        method: 'get',
+        url: `${API_URL}/extensions_manifest`,
+        timeout: 3000,
+      })
+        .then((response) => {
+          const manifest = response.data
+          this.manifest = manifest
+        })
+        .catch((error) => {
+          notifier.pushBackError('EXTENSIONS_MANIFEST_FETCH_FAIL', error)
+        })
+    },
+    async fetchInstalledExtensions(): Promise<void> {
+      await back_axios({
+        method: 'get',
+        url: `${API_URL}/installed_extensions`,
+        timeout: 30000,
+      })
+        .then((response) => {
+          const extensions = response.data
+          this.installed_extensions = extensions
+          this.dockers_fetch_done = true
+        })
+        .catch((error) => {
+          notifier.pushBackError('EXTENSIONS_INSTALLED_FETCH_FAIL', error)
+        })
+    },
+    showModal(extension: ExtensionData) {
+      this.show_dialog = true
+      this.selected_extension = extension
+    },
+    async install(tag: string) {
+      await axios.post(`${API_URL}/extension/install`, {
+        name: this.selected_extension?.docker,
+        tag,
+        enabled: true,
+        permissions: JSON.stringify(this.selected_extension?.versions[tag].permissions),
+
+      })
+        .then(() => {
+          this.show_dialog = false
+          this.fetchInstalledExtensions()
+        })
+        .catch((error) => {
+          notifier.pushBackError('EXTENSIONS_INSTALL_FAIL', error)
+        })
+    },
+    async uninstall(extension: ExtensionData) {
+      await axios.post(`${API_URL}/extension/uninstall`, null, {
+        params: {
+          extension_name: extension.name,
+        },
+      })
+        .then(() => {
+          this.fetchInstalledExtensions()
+        })
+        .catch((error) => {
+          notifier.pushBackError('EXTENSIONS_UNINSTALL_FAIL', error)
+        })
+    },
+    installedVersion(): string | undefined {
+      const extension_name = this.selected_extension?.docker
+      if (!extension_name) {
+        return undefined
+      }
+      const versions = this.installed_extensions
+        .find((extension) => extension.name === extension_name)?.versions
+      return versions?.length ? versions[0].tag : undefined
```

Comment:
you can use cosmos `first` function
```suggestion
      return versions?.first()?.tag
```

Reply:
nice